src/GovUk.Education.ExploreEducationStatistics.Data.Services/LocationPredicateBuilder.cs

```diff
@@ -8,62 +8,43 @@
 using GovUk.Education.ExploreEducationStatistics.Common.Model.Data.Query;
 using GovUk.Education.ExploreEducationStatistics.Data.Model;
 using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
-using GovUk.Education.ExploreEducationStatistics.Data.Model.Query;
 
 namespace GovUk.Education.ExploreEducationStatistics.Data.Services
 {
-    public static class ObservationPredicateBuilder
+    public static class LocationPredicateBuilder
     {
-        public static Expression<Func<Observation, bool>> Build(SubjectMetaQueryContext query)
+        public static Expression<Func<Location, bool>> Build(
+            IEnumerable<Guid>? locationIds,
+            LocationQuery? locationCodes)
         {
-            var predicate = PredicateBuilder.True<Observation>()
-                .AndAlso(observation => observation.SubjectId == query.SubjectId);
+            var predicate = PredicateBuilder.True<Location>();
 
-            if (query.TimePeriod != null)
+            if (locationIds != null)
             {
-                var timePeriodRange = TimePeriodUtil.Range(query.TimePeriod);
-
-                var subPredicate = PredicateBuilder.False<Observation>();
-
-                foreach (var tuple in timePeriodRange)
-                {
-                    var year = tuple.Year;
-                    var timeIdentifier = tuple.TimeIdentifier;
-
-                    subPredicate = subPredicate.Or(observation =>
-                        observation.Year == year && observation.TimeIdentifier == timeIdentifier);
-                }
-
-                predicate = predicate.AndAlso(subPredicate);
-            }
-
-            if (query.LocationIds != null)
-            {
-                predicate = predicate.AndAlso(observation =>
-                    query.LocationIds.Contains(observation.LocationId));
+                predicate = predicate.AndAlso(location =>
+                    locationIds.Contains(location.Id));
             }
 
             // TODO EES-3068 Migrate Location codes to ids in old Datablocks to remove this support for Location codes
-            if (query.Locations != null)
+            if (locationCodes != null)
             {
-                if (query.Locations.GeographicLevel != null)
+                if (locationCodes.GeographicLevel != null)
                 {
-                    predicate = predicate.AndAlso(observation =>
-                        observation.GeographicLevel == query.Locations.GeographicLevel);
+                    predicate = predicate.AndAlso(location => location.GeographicLevel == locationCodes.GeographicLevel);
```

Comment:
Not really for this PR but I looked at all data blocks and permalinks and I don't think `GeographicLevel` has ever been allowed to be set as a query param so I think we could drop it in future.